### PR TITLE
Binge mode

### PR DESCRIPTION
**Pull request by Dankni95**
Heya!

Had some free time so i coded `binge mode`. This takes in multiple episodes and queues them while the first video is playing.
Every time the player closes the next video will be played.

The download mode code was really helpful here!

The inspiration is taken from feature request for ani-cli, this is good way to differentiate us: https://github.com/pystardust/ani-cli/issues/102

Is this something we want to implement? if so, please test this before merging :smile: 

Daniel

### CODE REVIEW

File: main.py [hunk 1]
```diff
@@ -21,28 +21,32 @@
...
                        required=False,
                        dest='delete',
                        action='store_true',
                        help='Delete your History.')
+my_parser.add_argument("-b",
+                        "--binge",
+                        required=False,
+                        dest="binge",
+                        action="store_true",
+                        help="Binge mode. Binge multiple episodes like so: first number-second_number (e.g. 1-3)",
```
Comment (sdaqo) [inline main.py:48]: typo: first_number

**Comment by sdaqo** [comment]
I like this, but could you add some sort of way to exit cleanly, for example: It gets the first episode, and you get a menu like you normaly do but only with one option and that is quit maybe [this](https://github.com/sdaqo/anipy-cli/blob/42430745fd8fc796fa329b7a1c4ac0bd2930d6e0/src/menu.py#L189) helps. And you could put the for loop that gets the remaining links in a separate thread.

File: main.py (hunk 1 above)
Comment (Dankni95) [reply main.py]: Thanks. Fixed!

**Comment by Dankni95** [comment]
> I like this, but could you add some sort of way to exit cleanly, for example: It gets the first episode, and you get a menu like you normaly do but only with one option and that is quit maybe [this](https://github.com/sdaqo/anipy-cli/blob/42430745fd8fc796fa329b7a1c4ac0bd2930d6e0/src/menu.py#L189) helps. And you could put the for loop that gets the remaining links in a separate thread.

Added way to exit "cleanly". I had to remove some prints in helper functions, i dont think they should be there. When using Threads they cause some problems. What do you think about these fixes?

**Comment by sdaqo** [comment]
Oh no that's plenty, its just that fetching the video links just takes some time, the embed links really don't take a long time so thats ok.

**Comment by Dankni95** [comment]
There is a fully threaded version i just added, if that one better, we can add it :smile: 
https://github.com/Dankni95/anipy-cli/commit/d642f968689976a7306e01c44e29b4012fdadcc5

**Comment by sdaqo** [comment]
Ok nice, could you open up another PR for this?

**Pull request merged**